Mathlib/MeasureTheory/Integral/MeanInequalities.lean

```diff
@@ -284,17 +285,12 @@ theorem lintegral_rpow_add_lt_top_of_lintegral_rpow_lt_top {p : ℝ} {f g : α 
             hp1
         rw [ENNReal.div_add_div_same, one_add_one_eq_two,
           ENNReal.div_self two_ne_zero ENNReal.coe_ne_top]
-      · rw [← lt_top_iff_ne_top]
-        refine ENNReal.rpow_lt_top_of_nonneg hp0 ?_
-        rw [one_div, ENNReal.inv_ne_top]
-        exact two_ne_zero
+      · finiteness
     _ < ⊤ := by
-      have h_two : (2 : ℝ≥0∞) ^ (p - 1) ≠ ⊤ :=
-        ENNReal.rpow_ne_top_of_nonneg (by simp [hp1]) ENNReal.coe_ne_top
       rw [lintegral_add_left', lintegral_const_mul'' _ (hf.pow_const p),
-        lintegral_const_mul' _ _ h_two, ENNReal.add_lt_top]
-      · exact ⟨ENNReal.mul_lt_top h_two.lt_top hf_top, ENNReal.mul_lt_top h_two.lt_top hg_top⟩
-      · exact (hf.pow_const p).const_mul _
+        lintegral_const_mul' _ _ (by finiteness), ENNReal.add_lt_top]
```

Comment:
regression?

Reply:
Actually, I was wrong: in turn for the `by finiteness`, I'm removing a hypothesis above which was proved in two lines. Overall, I consider this an improvement.